File: Lua/conversor.lua
function decimalParaBinario(decimal)
    if decimal == 0 then
        return "0"
    end

    local binario = ""
    local numero = decimal
    while numero > 0 do
        binario = (numero % 2) .. binario
        numero = math.floor(numero / 2)
    end
    return binario
end

io.write("Digite um numero decimal: ")
local decimal = tonumber(io.read())
print("Binario: " .. decimalParaBinario(decimal))

File: Javascript/conversor.js
function decimalParaBinario(decimal) {
    if(decimal===0) return "0";

    let binario = "";
    let numero = decimal;
    while(numero>0){
        binario = (numero%2) + binario;
        numero = Math.floor(numero/2);
    }
    return binario;
}
const decimal = parseInt(prompt("Digite um numero decimal:"));
console.log("Binario: " + decimalParaBinario(decimal));
